src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure {
+    private double width;
+    private double height;
+
+    public Rectangle(Color color, double width, double height) {
+        super(color);
+        this.width = width;
+        this.height = height;
+    }
+
+    @Override
+    public double getArea() {
+        return width * height;
+    }
+
+    @Override
+    public void print() {
+        System.out.println("Figure: rectangle, area: " + getArea() + " sq. units, width: "
+                + width + " units, height: " + height + " units, color: "
+                + new ColorSupplier().getRandomColor());

```

Comment:
Instead of directly printing the color information in the `print()` method, you should use the `getColor()` method from the superclass to get the color that was already set in the constructor.